src/Operation.idr

```diff
@@ -0,0 +1,24 @@
+||| Module with definition and basic properties of extensional n-ary operations
+module Operation
+
+import public Data.Fun
+import public Data.Vect
+import public Setoid
+
+%access public export
+%default total
+
+using (arity : Nat)
+    CongruenceTy : (argsTys : Vect arity Setoid) -> (resTy : Setoid) -> Fun (map Carrier argsTys) (Carrier resTy) -> Type
+    CongruenceTy argsTys resTy f = go argsTys f f where
+        go : (argsTys : Vect arity Setoid) -> (f, g : Fun (map Carrier argsTys) (Carrier resTy)) -> Type
```

Comment:
I guess it's better to name it as `construct`, because it constructs the type of function.